keyboards/kprepublic/bm16v2/info.json

```diff
@@ -0,0 +1,81 @@
+{
+    "manufacturer": "kprepublic",
+    "keyboard_name": "BM16v2",
+    "maintainer": "qmk",
+    "bootloader": "stm32duino",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true
+    },
+    "matrix_pins": {
+        "cols": ["B1", "A3", "B0", "B4"],
+        "rows": ["A8", "A9", "B5", "B3"]
+    },
+    "processor": "STM32F103", // GD32F303CCT6
```

Comment:
Note that this configuration limits the usable flash size for the firmware to 56K due to the default `MCU_LDSCRIPT = STM32F103x8_stm32duino_bootloader`; although I'm not sure that KPrepublic won't “optimize” the hardware in some further revision to use a smaller chip.

Also did you check that the EEPROM emulation behaves properly with this configuration?  F103x8/xB chips have 1K flash sectors, but F103xC and larger chips have 2K flash sectors.  Apparently the default EEPROM driver for STM32F1xx is now `wear_leveling`/`embedded_flash`, and the EFL code in `lib/chibios/os/hal/ports/STM32/STM32F1xx/hal_efl_lld.c` does not perform any autodetection for the flash page size, therefore it probably does not work correctly when the real chip does not match the compile-time configuration.  (This problem does not affect the stm32duino bootloader, because it reads the flash size register and detects the flash sector size from that info, so it should work with all F103 chips with single bank flash.)

Adding `board.h` with
```c
#pragma once
#include_next <board.h>
#undef STM32F103xB
#define STM32F103xE
```
may be enough to fix the problem (assuming that KPrepublic would stick to using the GD32F303CCT6 chips on this board and won't downgrade to GD32F103CBT6 or something like that).

(Specifying `STM32F103xE` should be safe, because the code in `platforms/chibios/drivers/wear_leveling/wear_leveling_efl.c` support flash size detection for STM32F1xx to determine the backing store location; but that code relies on the ChibiOS EFL driver to get the flash sector size, and that part can be configured only during compile time on STM32F1xx.)

Reply:
EEPROM does seem to work; the RGB Matrix settings persist and it boots up with the correct effect and HSV after changing it.